index.js

```diff
@@ -256,15 +256,26 @@ Requests.type = 'XHR' === Requests.method ? (function detect() {
     type = types.pop();
 
     xhr = Requests.XHR();
-    xhr.open('get', 'http://localhost/', true);
-    prop = type.replace(/-/g, '');
 
     //
-    // We can only set the `responseType` after we've opened the connection or
-    // FireFox will throw an error and according to the spec only async requests
-    // can use this, which is fine as we force that by default.
+    // Older versions of Firefox/IE11 will throw an error because previous
+    // version of the specification do not support setting `responseType`
+    // before the request is opened. Thus, we open the request here.
+    //
+    // Note that `open()` does not actually open any connections; it just
+    // initializes the request object.
     //
     try {
+      // Try opening a request to current page.
+      xhr.open('get', '/', true);
```

Comment:
@Swaagie I've just tried and it seems to work fine. See https://saucelabs.com/beta/tests/585dcfa07fbb46df8bb58cd49b37edfa/watch

If you don't have flash installed (I don't), see https://assets.saucelabs.com/jobs/585dcfa07fbb46df8bb58cd49b37edfa/0002screenshot.png

This is the script http://tempjs.org/files/12c0dea170139988bd807d980cf73c0031f78926/index.js